plugin/session_buffer.py

```diff
@@ -56,24 +57,20 @@ class SessionBuffer:
     """
     Holds state per session per buffer.
 
-    It stores the filename, handles document synchronization for the buffer, and stores/receives diagnostics for the
+    It stores the URI, handles document synchronization for the buffer, and stores/receives diagnostics for the
     buffer. The diagnostics are then published further to the views attached to this buffer. It also maintains the
     dynamically registered capabilities applicable to this particular buffer.
     """
 
-    def __init__(self, session_view: SessionViewProtocol, buffer_id: int, language_id: str) -> None:
+    def __init__(self, session_view: SessionViewProtocol, buffer_id: int, uri: DocumentUri) -> None:
         view = session_view.view
-        file_name = view.file_name()
-        if not file_name:
-            raise ValueError("missing filename")
         self.opened = False
         # Every SessionBuffer has its own personal capabilities due to "dynamic registration".
         self.capabilities = Capabilities()
         self.session = session_view.session
         self.session_views = WeakSet()  # type: WeakSet[SessionViewProtocol]
         self.session_views.add(session_view)
-        self.file_name = file_name
-        self.language_id = language_id
+        self.uri = uri
```

Comment:
It's not really clear why this class has both `self.uri` and `self.get_uri()` and what's the difference between those. Maybe renaming this would help? Like `last_known_uri` maybe? It would also help with accidentally using the wrong wrong.